catroid/src/main/java/org/catrobat/catroid/content/bricks/PlaySoundBrick.java

```diff
@@ -24,252 +24,151 @@
+					sound = getSoundByName(spinnerAdapter.getItem(position));
+				}
+			}
 
-		@Override
-		public void unregisterDataSetObserver(DataSetObserver paramDataSetObserver) {
-			spinnerAdapter.unregisterDataSetObserver(paramDataSetObserver);
-		}
+			@Override
+			public void onNothingSelected(AdapterView<?> parent) {
+			}
+		});
+		spinner.setSelection(spinnerAdapter.getPosition(sound != null ? sound.getName() : null));
 
-		@Override
-		public int getCount() {
-			return spinnerAdapter.getCount();
+		if (wait) {
+			((TextView) view.findViewById(R.id.brick_play_sound_label)).setText(R.string.brick_play_sound_and_wait);
 		}
 
-		@Override
-		public Object getItem(int paramInt) {
-			return spinnerAdapter.getItem(paramInt);
-		}
+		return view;
+	}
 
-		@Override
-		public long getItemId(int paramInt) {
-			SoundInfo currentSound = spinnerAdapter.getItem(paramInt);
-			if (!currentSound.getName().equals(context.getString(R.string.new_broadcast_message))) {
-				oldSelectedSound = currentSound;
+	@Override
+	public boolean onNewOptionInDropDownClicked(View v) {
+		spinnerSelectionBuffer = spinner.getSelectedItemPosition();
+		new NewSoundDialogFragment(this,
+				ProjectManager.getInstance().getCurrentScene(),
+				ProjectManager.getInstance().getCurrentSprite()) {
+
+			@Override
+			public void onCancel(DialogInterface dialog) {
+				super.onCancel(dialog);
+				spinner.setSelection(spinnerSelectionBuffer);
 			}
-			return spinnerAdapter.getItemId(paramInt);
-		}
-
-		@Override
-		public boolean hasStableIds() {
-			return spinnerAdapter.hasStableIds();
-		}
+		}.show(((Activity) v.getContext()).getFragmentManager(), NewSoundDialogFragment.TAG);
+		return false;
+	}
 
-		@Override
-		public View getView(int paramInt, View paramView, ViewGroup paramViewGroup) {
-			if (isTouchInDropDownView) {
-				isTouchInDropDownView = false;
-				if (paramInt == 0) {
-					showNewSoundDialog((Activity) context);
-				}
-			}
-			return spinnerAdapter.getView(paramInt, paramView, paramViewGroup);
-		}
+	@Override
+	public void addItem(SoundInfo item) {
+		ProjectManager.getInstance().getCurrentSprite().getSoundList().add(item);
+		spinnerAdapter.add(item.getName());
+		sound = item;
+		spinner.setSelection(spinnerAdapter.getPosition(item.getName()));
+	}
 
-		@Override
-		public int getItemViewType(int paramInt) {
-			return spinnerAdapter.getItemViewType(paramInt);
-		}
+	@Override
+	public View getPrototypeView(Context context) {
+		View view = View.inflate(context, R.layout.brick_play_sound, null);
+		spinner = view.findViewById(R.id.playsound_spinner);
 
-		@Override
-		public int getViewTypeCount() {
-			return spinnerAdapter.getViewTypeCount();
-		}
+		spinnerAdapter = new SpinnerAdapterWithNewOption(context, getSoundNames());
+		spinner.setAdapter(spinnerAdapter);
+		spinner.setSelection(spinnerAdapter.getPosition(sound != null ? sound.getName() : null));
 
-		@Override
-		public boolean isEmpty() {
-			return spinnerAdapter.isEmpty();
+		if (wait) {
```

Comment:
yes, I did think about that too. In fact I was going to change the implementation so that it becomes even more adaptive. I'll have a look at it again when I have time.